File: backend-java/src/main/java/com/example/calorie/body/BodyRecordController.java
package com.example.calorie.body;

import com.example.calorie.body.dto.BodyDtos.BodyRecordRequest;
import com.example.calorie.body.dto.BodyDtos.BodyRecordResponse;
import jakarta.validation.Valid;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.HttpStatus;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;
import java.util.List;

@RestController
@RequestMapping("/api/body-records")
public class BodyRecordController {

    private final BodyRecordService service;

    public BodyRecordController(BodyRecordService service) {
        this.service = service;
    }

    @GetMapping
    public List<BodyRecordResponse> list(
            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate from,
            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate to,
            @AuthenticationPrincipal Long userId) {
        return service.list(userId, from, to);
    }

    @PutMapping("/{date}")
    public BodyRecordResponse put(
            @PathVariable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date,
            @Valid @RequestBody BodyRecordRequest request,
            @AuthenticationPrincipal Long userId) {
        return service.put(userId, date, request);
    }

    @DeleteMapping("/{date}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(
            @PathVariable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date,
            @AuthenticationPrincipal Long userId) {
        service.delete(userId, date);
    }
}

File: backend-kotlin/src/main/kotlin/com/example/calorie/body/BodyRecordController.kt
package com.example.calorie.body

import com.example.calorie.body.dto.BodyRecordRequest
import com.example.calorie.body.dto.BodyRecordResponse
import jakarta.validation.Valid
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.http.HttpStatus
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate

@RestController
@RequestMapping("/api/body-records")
class BodyRecordController(private val service: BodyRecordService) {

    @GetMapping
    fun list(
        @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) from: LocalDate,
        @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) to: LocalDate,
        @AuthenticationPrincipal userId: Long,
    ): List<BodyRecordResponse> = service.list(userId, from, to)

    @PutMapping("/{date}")
    fun put(
        @PathVariable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) date: LocalDate,
        @Valid @RequestBody request: BodyRecordRequest,
        @AuthenticationPrincipal userId: Long,
    ): BodyRecordResponse = service.put(userId, date, request)

    @DeleteMapping("/{date}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun delete(
        @PathVariable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) date: LocalDate,
        @AuthenticationPrincipal userId: Long,
    ) = service.delete(userId, date)
}
